template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/hbase/HbaseCloudStorageServiceConfigProvider.java

```diff
@@ -60,7 +61,7 @@ public boolean isConfigurationNeeded(CmTemplateProcessor cmTemplateProcessor, Te
                 entitlementService.sdxHbaseCloudStorageEnabled(accountId);
         return source.getFileSystemConfigurationView().isPresent()
                 && cmTemplateProcessor.isRoleTypePresentInService(getServiceType(), getRoleTypes())
-                && (!datalakeCluster || (is727OrNewer && sdxHbaseCloudStorageEnabled));
+                && (!datalakeCluster || (is727OrNewer && (source.getCloudPlatform().equals(CloudPlatform.AWS) || sdxHbaseCloudStorageEnabled)));
```

Comment:
I know `source.getCloudPlatform()` is not supposed to be `null`, but it is nevertheless safer to invoke `equals()` on `CloudPlatform.AWS`, so swapping it with `source.getCloudPlatform()`. Actually, since we are talking about enum constants that are singletons, the simple reference equality `==` would be also appropriate.

Reply:
agree. switched order